Document launching child proc using dotnet-trace and dotnet-counters (#21193)
dotnet/docs: docs/core/diagnostics/dotnet-counters.md

### Synopsis

```console

> dotnet-counters collect [-h|--help] [-p|--process-id] [--refreshInterval] [--counters] [counter_list] [--format] [-o|--output] [-- <command>]

Reviewer: ```suggestion
dotnet-counters collect [-h|--help] [-p|--process-id] [--refreshInterval] [--counters] [--format] [-o|--output] [-- <command>]
```
I suggest we document it as if --counters was required to specify the list. If people don't know about the secret back-compatible syntax I don't think they benefit from learning it.

Reviewer 2: ```suggestion
dotnet-counters collect [-h|--help] [-p|--process-id] [--refreshInterval] [--counters <COUNTERS>] [--format] [-o|--output] [-- <command>]
```

  The number of seconds to delay between updating the displayed counters

> - **`--counters`**

Reviewer: ```suggestion
- **`--counters <COUNTERS>`**
```

>   A comma-separated list of counters. Counters can be specified `provider_name[:counter_name]`. If the `provider_name` is used without a qualifying list of counters, then all counters from the provider are shown. To discover provider and counter names, use the [dotnet-counters list](#dotnet-counters-list) command.

Author: Nope, this is new. The old syntax is still supported, but I'm removing it from the docs in favor of the new syntax.

>   A comma-separated list of counters. Counters can be specified `provider_name[:counter_name]`. If the `provider_name` is used without a qualifying list of counters, then all counters from the provider are shown. To discover provider and counter names, use the [dotnet-counters list](#dotnet-counters-list) command.

Reviewer 3: Have these been released? Otherwise we'd end up with customers having tools that don't support this.

>   A comma-separated list of counters. Counters can be specified `provider_name[:counter_name]`. If the `provider_name` is used without a qualifying list of counters, then all counters from the provider are shown. To discover provider and counter names, use the [dotnet-counters list](#dotnet-counters-list) command.

Author: Not yet - which is why I mentioned in the PR description that we should only merge this after the tools have been released :)

>   A comma-separated list of counters. Counters can be specified `provider_name[:counter_name]`. If the `provider_name` is used without a qualifying list of counters, then all counters from the provider are shown. To discover provider and counter names, use the [dotnet-counters list](#dotnet-counters-list) command.

Reviewer 3: Ah. The diff viewer didn't list the PR description. That's my bad for not reading.

  The name of the output file.

- **`-- <command>` (for target applications running .NET 5.0 or later only)**

>   Putting `--` followed by a command to start a .NET 5.0 (or later) application after the `collect` command launches the given command as a child process and starts collecting metric from the target process. This is useful for getting metrics from runtime startup (before `Main` runs) and can be used to diagnose or monitor issues that happen early in the application startup.

Reviewer 3: ```suggestion
  After the collection configuration parameters, the user can append `--` followed by a command to start a .NET application with at least a 5.0 runtime. `dotnet-counters` will launch a process with the provided command and collect the requested metrics. This is often useful to collect metrics for the application's startup path and can be used to diagnose or monitor issues that happen early before or shortly after the main entrypoint.
```

### Synopsis

```console

> dotnet-counters monitor [-h|--help] [-p|--process-id] [--refreshInterval] [counter_list] [-- <command>]

Reviewer: ```suggestion
dotnet-counters monitor [-h|--help] [-p|--process-id] [--refreshInterval] [--counters] [-- <command>]
```

Reviewer 2: ```suggestion
dotnet-counters monitor [-h|--help] [-p|--process-id] [--refreshInterval] [--counters <counters-list>] [-- <command>]
```

  The number of seconds to delay between updating the displayed counters

- **`--counters`**

  A comma-separated list of counters. Counters can be specified `provider_name[:counter_name]`. If the `provider_name` is used without a qualifying list of counters, then all counters from the provider are shown. To discover provider and counter names, use the [dotnet-counters list](#dotnet-counters-list) command.

 **`-- <command>` (for target applications running .NET 5.0 or later only)**

>   Prefixing the `collect` command with `--` followed by a command executes the given command as a child process and starts collecting metric from the target process. This is useful for getting metrics from runtime startup (before `Main` runs) and can be used to diagnose or monitor issues that happen early in the application startup.

Reviewer 3: Hm. Not sure if prefixing is the right word here. Probably we should be consistent in the places where we document this.

  Press p to pause, r to resume, q to quit.
    Status: Running

  [System.Runtime]
      CPU Usage (%)                                 24
      GC Heap Size (MB)                            811
  ```

- Monitor `EventCounter` values from user-defined `EventSource`. For more information, see [Tutorial: Measure performance using EventCounters in .NET Core](event-counter-perf.md).

  ```console
  > dotnet-counters monitor --process-id 1902 --counters Samples-EventCounterDemos-Minimal

  Press p to pause, r to resume, q to quit.
      request                                      100
  ```

- Launch `my-aspnet-server.exe` and monitor the # of assemblies loaded from its startup (.NET 5.0 or later only):

  ```console
  > dotnet-counters monitor --counters System.Runtime[assembly-count] -- my-aspnet-server.exe

  Press p to pause, r to resume, q to quit.
    Status: Running

  [System.Runtime]
      Number of Assemblies Loaded                   24
  ```
  
- Launch `my-aspnet-server.exe` with `arg1` and `arg2` as command-line arguments and monitor its working set and GC heap size from its startup (.NET 5.0 or later only):

  ```console

>   > dotnet-counters monitor --counters System.Runtime[working-set,gc-heap-size]

Reviewer 3: ```suggestion
  > dotnet-counters monitor --counters System.Runtime[working-set,gc-heap-size] -- my-aspnet-server.exe arg1 arg2
```

  The number of seconds to delay between updating the displayed counters

> - **`--counters`**

Reviewer: ```suggestion
- **`--counters <COUNTERS>`**
```

 **`-- <command>` (for target applications running .NET 5.0 or later only)**

  After the collection configuration parameters, the user can append `--` followed by a command to start a .NET application with at least a 5.0 runtime. `dotnet-counters` will launch a process with the provided command and monitor the requested metrics. This is often useful to collect metrics for the application's startup path and can be used to diagnose or monitor issues that happen early before or shortly after the main entrypoint.

  > [!NOTE]
  > Using this option monitors the first .NET 5.0 process that communicates back to the tool, which means if your command launches multiple .NET applications, it will only collect the first app. Therefore, it is recommended you use this option on self-contained applications, or using the `dotnet exec <app.dll>` option.

### Examples

- Monitor all counters from `System.Runtime` at a refresh interval of 3 seconds:

  ```console

>   > dotnet-counters monitor --process-id 1902  --refresh-interval 3 System.Runtime

Reviewer 2: If we are going to recommend using the `--counters <COUNTERS>` convention, we should update the examples too.
